### PR TITLE
fix: api key improvement

### LINKED ISSUE #708: API Key improvements
Improve API Key UX with:
UI 
Settings -> API Key
1. Improve UX with CRUD
2. Add name and details ("api key for team X")

Alerts
3. Add the ability to filter by api key in the UI (or make sure it works)

Backend
4. When pushing alerts via api key, save the relation so it can be later filtered

### PR DESCRIPTION
**Pull request by Min2who**
Closes #708 

## 📑 Description
This is work in progress

### CODE REVIEW

**Comment by Min2who** [comment]
can yu reviwed the backend changes 
cc @shahargl

File: keep/api/models/apikey.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from typing import Optional
+
+from pydantic import BaseModel, Extra
+
+class APIKeyDTO(BaseModel, extra=Extra.ignore):
+    key_name: str
+    description: str
+    created_at: str
```
Comment (shahargl) [inline keep/api/models/apikey.py:8]: Add also:
1. role
2. last used (we will update when the api key is used)

File: keep/api/models/db/apikey.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from datetime import datetime
+
+from sqlmodel import Field, SQLModel
+
+# THIS IS ONLY FOR SINGLE TENANT (self-hosted) USAGES
+from keep.api.core.dependencies import SINGLE_TENANT_UUID
+
+class APIKey(SQLModel, table=True):
```
Comment (shahargl) [inline keep/api/models/db/apikey.py:8]: you don't need a new table. there is already tenantapikey, just use it.

**Review by shahargl — CHANGES_REQUESTED** [review]
see comments

**Comment by Min2who** [comment]
> see comments

@shahargl added the changes

**Comment by Min2who** [comment]
> Once you’ll have the UI I’ll be able to test it @Min2who

@shahargl really sorry for late reply . you can open the bounty to other i have some college work that pop up and will be occupied with it upto 3 weeks

**Pull request closed without merging**